### ISSUE TITLE
Komodo hanging and freezing

### ISSUE THREAD

**Issue by newtonsspawn** [Komodo/KomodoEdit#3611]
### Short Summary
Komodo frequently hangs when accessing menus (specifically accessing a menu a second time), attempting to save files,  or right-clicking items on the screen.

I upgraded from Komodo 10 to 11, and then installed the latest nighty-build version with the hopes this issue would be resolved. I even waited for a MacOS update and installed the public beta in the hopes that the issue would resolve itself but to no avail.

HELP!!!

### Steps to Reproduce
Basically almost anything I do in Komodo causes the issue. Anytime a dialogue box opens up to save or open a file, or when I right(option)-click anything one the screen the software hangs or completely freezes and I have to do a Force Quit.

### Platform Information

Komodo IDE 10, 11, and latest nightly build all have same issue.
Mac OS Mojave10.14 and 10.14.1 Beta (18B50c)

**Comment by Naatan** [comment]
Could you see if the troubleshooting steps here help you?

http://docs.activestate.com/komodo/11/manual/trouble.html#artifacts

**Comment by vbertola** [comment]
I have a problem which may be related - Komodo Edit (both the 10.2 I was using, and the latest 11.1 just downloaded) hangs completely on my Mac OS X 10.14 when I try to save a new file; the Save As dialog pops up, but as soon as I click on a couple of subdirectories it freezes and then it's completely stuck, I can't get out of Komodo unless I just kill the process manually. It's become completely impossible to use it!

This is what I get in system.log:

Nov 16 12:11:48 grigione komodo[2276]: DEPRECATED USE in libdispatch client: dispatch source activated with no event handler set; set a breakpoint on _dispatch_bug_deprecated to debug
Nov 16 12:11:58 grigione komodo[2276]: objc[2276]: Class FIFinderSyncExtensionHost is implemented in both /System/Library/PrivateFrameworks/FinderKit.framework/Versions/A/FinderKit (0x7fffa45a81d0) and /System/Library/PrivateFrameworks/FileProvider.framework/OverrideBundles/FinderSyncCollaborationFileProviderOverride.bundle/Contents/MacOS/FinderSyncCollaborationFileProviderOverride (0x12b1bfdc8). One of the two will be used. Which one is undefined.
Nov 16 12:11:59 grigione komodo[2276]: assertion failed: 18A391: libxpc.dylib + 92361 [0A8747D1-33AA-37E1-B97A-BA9B95FE4E8C]: 0x89
Nov 16 12:11:59 grigione com.apple.xpc.launchd[1] (com.seafile.seafile-client.findersync.732CFA0B-88A9-4CA7-B928-213C59B65A0C[2281]): Caller wanted oneshot behavior for pre-existing instance: caller = komodo


This is pystderr.log:

2018-11-16 12:11:48,331] [INFO] Startup: Welcome to Komodo Edit 11.1.0 build 18196 (platform macosx, running on Darwin 18.0.0 version Darwin Kernel Version 18.0.0: Wed Aug 22 20:13:40 PDT 2018; root:xnu-4903.201.2~1/RELEASE_X86_64)
[2018-11-16 12:11:48,332] [INFO] Startup: /Applications/Komodo Edit 11.app/Contents/MacOS/komodo built on Wed May 23 00:03:29 2018
[2018-11-16 12:11:48,510] [WARNING] koInitService: Unable to determine the current locale settings, defaulting to mac-roman
[2018-11-16 12:11:48,779] [WARNING] console-logger: mutating the [[Prototype]] of an object will cause your code to run very slowly; instead create the object with the correct initial [[Prototype]] value using Object.create (1) in resource://gre/modules/Preferences.jsm:381
[2018-11-16 12:11:49,191] [WARNING] root: [object Object].logging has been converted to a CommonJS module; use require("ko/logging") instead (since Komodo 9.0.0a1).
    @chrome://komodo/content/komodo.js:48:1

[2018-11-16 12:11:49,622] [ERROR] console-logger: bad script XDR magic number (0) in :0
Traceback (most recent call last):
  File "", line 0, in 

[2018-11-16 12:11:50,221] [WARNING] ko.widgets: Can't load 'ui.tabs.sidepanes.state' from 'prefpath': 
[2018-11-16 12:11:50,887] [INFO] koInitService: Adding pre startup service for 'koFileStatusService': '@activestate.com/koFileStatusService;1'
[2018-11-16 12:11:50,908] [INFO] elastic_tabstops: Elastic tabstops loaded.
[2018-11-16 12:11:51,437] [ERROR] console-logger: Exception while registering Focus Mode (0) in resource://gre/modules/addons/XPIProvider.jsm -> file:///Applications/Komodo%20Edit%2011.app/Contents/Resources/extensions/focusmode@activestate.com/bootstrap.js:11
Traceback (most recent call last):
  File "resource://gre/modules/addons/XPIProvider.jsm -> file:///Applications/Komodo%20Edit%2011.app/Contents/Resources/extensions/focusmode@activestate.com/bootstrap.js", line 11, in 

[2018-11-16 12:11:51,442] [ERROR] console-logger: [object Object] (0) in resource://gre/modules/addons/XPIProvider.jsm -> file:///Applications/Komodo%20Edit%2011.app/Contents/Resources/extensions/focusmode@activestate.com/bootstrap.js:12
Traceback (most recent call last):
  File "resource://gre/modules/addons/XPIProvider.jsm -> file:///Applications/Komodo%20Edit%2011.app/Contents/Resources/extensions/focusmode@activestate.com/bootstrap.js", line 12, in 

[2018-11-16 12:11:51,699] [WARNING] console-logger: mutating the [[Prototype]] of an object will cause your code to run very slowly; instead create the object with the correct initial [[Prototype]] value using Object.create (1) in file:///Applications/Komodo%20Edit%2011.app/Contents/Resources/components/koamAddonManager.js:138
[2018-11-16 12:11:52,806] [WARNING] koScintillaSchemeService: applyScheme:: no indicator for name 'collab_remote_cursor_4'
[2018-11-16 12:11:52,807] [WARNING] koScintillaSchemeService: applyScheme:: no indicator for name 'collab_remote_cursor_5'
[2018-11-16 12:11:52,807] [WARNING] koScintillaSchemeService: applyScheme:: no indicator for name 'collab_remote_cursor_2'
[2018-11-16 12:11:52,807] [WARNING] koScintillaSchemeService: applyScheme:: no indicator for name 'collab_remote_cursor_3'
[2018-11-16 12:11:52,807] [WARNING] koScintillaSchemeService: applyScheme:: no indicator for name 'collab_remote_cursor_1'
[2018-11-16 12:11:52,807] [WARNING] koScintillaSchemeService: applyScheme:: no indicator for name 'collab_remote_change'
[2018-11-16 12:11:52,808] [WARNING] koScintillaSchemeService: applyScheme:: no indicator for name 'spelling_error'
[2018-11-16 12:11:52,808] [WARNING] koScintillaSchemeService: applyScheme:: no indicator for name 'collab_local_change'
[2018-11-16 12:11:53,073] [WARNING] views: Inconsistent view count in view_closed event
[2018-11-16 12:11:53,083] [WARNING] koScintillaSchemeService: applyScheme:: no indicator for name 'collab_remote_cursor_4'
[2018-11-16 12:11:53,084] [WARNING] koScintillaSchemeService: applyScheme:: no indicator for name 'collab_remote_cursor_5'
[2018-11-16 12:11:53,084] [WARNING] koScintillaSchemeService: applyScheme:: no indicator for name 'collab_remote_cursor_2'
[2018-11-16 12:11:53,084] [WARNING] koScintillaSchemeService: applyScheme:: no indicator for name 'collab_remote_cursor_3'
[2018-11-16 12:11:53,084] [WARNING] koScintillaSchemeService: applyScheme:: no indicator for name 'collab_remote_cursor_1'
[2018-11-16 12:11:53,084] [WARNING] koScintillaSchemeService: applyScheme:: no indicator for name 'collab_remote_change'
[2018-11-16 12:11:53,085] [WARNING] koScintillaSchemeService: applyScheme:: no indicator for name 'spelling_error'
[2018-11-16 12:11:53,085] [WARNING] koScintillaSchemeService: applyScheme:: no indicator for name 'collab_local_change'
[2018-11-16 12:11:53,126] [ERROR] CT::margin.js: couldn't get the insert-color
-- EXCEPTION START --
TypeError: this.view.scheme.getInterfaceStyleRaw is not a function
+ stack
    exports.MarginController.prototype._initMarkerStyles@chrome://trackchanges/content/margin.js:59:32
    refreshMarginProperies@chrome://trackchanges/content/margin.js:120:9
    MarginController@chrome://trackchanges/content/margin.js:44:5
    exports.ChangeTracker.prototype.changeTrackingOn@chrome://trackchanges/content/tracker.js:63:19
    ChangeTracker@chrome://trackchanges/content/tracker.js:27:5
    this.onViewOpenedHandler@chrome://trackchanges/content/trackchanges.js:88:26
    xtk.domutils.fireEvent@chrome://xtk/content/domutils.js:187:5
    set_koDoc@vieweditor:297:17
    initWithDocument@vieweditor:410:13
    createViewFromDocument@viewlistbase:133:17
    viewManager.prototype._doNewView@chrome://komodo/content/views.js:267:16
    viewManager.prototype.doNewViewAsync/<@chrome://komodo/content/views.js:286:20
-- EXCEPTION END --
[2018-11-16 12:11:53,137] [ERROR] CT::margin.js: couldn't get the delete-color
-- EXCEPTION START --
TypeError: this.view.scheme.getInterfaceStyleRaw is not a function
+ stack
    exports.MarginController.prototype._initMarkerStyles@chrome://trackchanges/content/margin.js:65:32
    refreshMarginProperies@chrome://trackchanges/content/margin.js:120:9
    MarginController@chrome://trackchanges/content/margin.js:44:5
    exports.ChangeTracker.prototype.changeTrackingOn@chrome://trackchanges/content/tracker.js:63:19
    ChangeTracker@chrome://trackchanges/content/tracker.js:27:5
    this.onViewOpenedHandler@chrome://trackchanges/content/trackchanges.js:88:26
    xtk.domutils.fireEvent@chrome://xtk/content/domutils.js:187:5
    set_koDoc@vieweditor:297:17
    initWithDocument@vieweditor:410:13
    createViewFromDocument@viewlistbase:133:17
    viewManager.prototype._doNewView@chrome://komodo/content/views.js:267:16
    viewManager.prototype.doNewViewAsync/<@chrome://komodo/content/views.js:286:20
-- EXCEPTION END --
[2018-11-16 12:11:53,141] [ERROR] CT::margin.js: couldn't get the change-color
-- EXCEPTION START --
TypeError: this.view.scheme.getInterfaceStyleRaw is not a function
+ stack
    exports.MarginController.prototype._initMarkerStyles@chrome://trackchanges/content/margin.js:71:33
    refreshMarginProperies@chrome://trackchanges/content/margin.js:120:9
    MarginController@chrome://trackchanges/content/margin.js:44:5
    exports.ChangeTracker.prototype.changeTrackingOn@chrome://trackchanges/content/tracker.js:63:19
    ChangeTracker@chrome://trackchanges/content/tracker.js:27:5
    this.onViewOpenedHandler@chrome://trackchanges/content/trackchanges.js:88:26
    xtk.domutils.fireEvent@chrome://xtk/content/domutils.js:187:5
    set_koDoc@vieweditor:297:17
    initWithDocument@vieweditor:410:13
    createViewFromDocument@viewlistbase:133:17
    viewManager.prototype._doNewView@chrome://komodo/content/views.js:267:16
    viewManager.prototype.doNewViewAsync/<@chrome://komodo/content/views.js:286:20
-- EXCEPTION END --
[2018-11-16 12:11:53,176] [WARNING] views: Inconsistent view count in view_opened event
[2018-11-16 12:11:53,583] [INFO] koInitService: Adding pre startup service for 'KoMemoryReporter': '@activestate.com/koMemoryReporter;1'
[2018-11-16 12:11:53,587] [INFO] koInitService: Adding pre startup service for 'koCommandmentService': '@activestate.com/koCommandmentService;1'
[2018-11-16 12:11:54,032] [WARNING] koScintillaSchemeService: applyScheme:: no indicator for name 'collab_remote_cursor_4'
[2018-11-16 12:11:54,032] [WARNING] koScintillaSchemeService: applyScheme:: no indicator for name 'collab_remote_cursor_5'
[2018-11-16 12:11:54,032] [WARNING] koScintillaSchemeService: applyScheme:: no indicator for name 'collab_remote_cursor_2'
[2018-11-16 12:11:54,032] [WARNING] koScintillaSchemeService: applyScheme:: no indicator for name 'collab_remote_cursor_3'
[2018-11-16 12:11:54,033] [WARNING] koScintillaSchemeService: applyScheme:: no indicator for name 'collab_remote_cursor_1'
[2018-11-16 12:11:54,033] [WARNING] koScintillaSchemeService: applyScheme:: no indicator for name 'collab_remote_change'
[2018-11-16 12:11:54,033] [WARNING] koScintillaSchemeService: applyScheme:: no indicator for name 'spelling_error'
[2018-11-16 12:11:54,034] [WARNING] koScintillaSchemeService: applyScheme:: no indicator for name 'collab_local_change'
[2018-11-16 12:11:54,064] [ERROR] CT::margin.js: couldn't get the insert-color
-- EXCEPTION START --
TypeError: this.view.scheme.getInterfaceStyleRaw is not a function
+ stack
    exports.MarginController.prototype._initMarkerStyles@chrome://trackchanges/content/margin.js:59:32
    refreshMarginProperies@chrome://trackchanges/content/margin.js:120:9
    MarginController@chrome://trackchanges/content/margin.js:44:5
    exports.ChangeTracker.prototype.changeTrackingOn@chrome://trackchanges/content/tracker.js:63:19
    ChangeTracker@chrome://trackchanges/content/tracker.js:27:5
    this.onViewOpenedHandler@chrome://trackchanges/content/trackchanges.js:88:26
    xtk.domutils.fireEvent@chrome://xtk/content/domutils.js:187:5
    set_koDoc@vieweditor:297:17
    initWithDocument@vieweditor:410:13
    createViewFromDocument@viewlistbase:133:17
    viewManager.prototype._doNewView@chrome://komodo/content/views.js:267:16
    viewManager.prototype.doNewViewAsync/<@chrome://komodo/content/views.js:286:20
-- EXCEPTION END --
[2018-11-16 12:11:54,068] [ERROR] CT::margin.js: couldn't get the delete-color
-- EXCEPTION START --
TypeError: this.view.scheme.getInterfaceStyleRaw is not a function
+ stack
    exports.MarginController.prototype._initMarkerStyles@chrome://trackchanges/content/margin.js:65:32
    refreshMarginProperies@chrome://trackchanges/content/margin.js:120:9
    MarginController@chrome://trackchanges/content/margin.js:44:5
    exports.ChangeTracker.prototype.changeTrackingOn@chrome://trackchanges/content/tracker.js:63:19
    ChangeTracker@chrome://trackchanges/content/tracker.js:27:5
    this.onViewOpenedHandler@chrome://trackchanges/content/trackchanges.js:88:26
    xtk.domutils.fireEvent@chrome://xtk/content/domutils.js:187:5
    set_koDoc@vieweditor:297:17
    initWithDocument@vieweditor:410:13
    createViewFromDocument@viewlistbase:133:17
    viewManager.prototype._doNewView@chrome://komodo/content/views.js:267:16
    viewManager.prototype.doNewViewAsync/<@chrome://komodo/content/views.js:286:20
-- EXCEPTION END --
[2018-11-16 12:11:54,073] [ERROR] CT::margin.js: couldn't get the change-color
-- EXCEPTION START --
TypeError: this.view.scheme.getInterfaceStyleRaw is not a function
+ stack
    exports.MarginController.prototype._initMarkerStyles@chrome://trackchanges/content/margin.js:71:33
    refreshMarginProperies@chrome://trackchanges/content/margin.js:120:9
    MarginController@chrome://trackchanges/content/margin.js:44:5
    exports.ChangeTracker.prototype.changeTrackingOn@chrome://trackchanges/content/tracker.js:63:19
    ChangeTracker@chrome://trackchanges/content/tracker.js:27:5
    this.onViewOpenedHandler@chrome://trackchanges/content/trackchanges.js:88:26
    xtk.domutils.fireEvent@chrome://xtk/content/domutils.js:187:5
    set_koDoc@vieweditor:297:17
    initWithDocument@vieweditor:410:13
    createViewFromDocument@viewlistbase:133:17
    viewManager.prototype._doNewView@chrome://komodo/content/views.js:267:16
    viewManager.prototype.doNewViewAsync/<@chrome://komodo/content/views.js:286:20
-- EXCEPTION END --


Can you do anything?

**Comment by Naatan** [comment]
@vbertola this should be fixed in the upcoming 11.1.1 release
